examples/rag/use_own_knowledge_dataset.py

```diff
@@ -0,0 +1,125 @@
+import os
+from functools import partial
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import List
+
+import faiss
+import torch
+from datasets import load_dataset
+
+from transformers import (
+    DPRContextEncoder,
+    DPRContextEncoderTokenizerFast,
+    RagRetriever,
+    RagSequenceForGeneration,
+    RagTokenizer,
+)
+
+
+torch.set_grad_enabled(False)
+device = "cuda" if torch.cuda.is_available() else "cpu"
+
+
+def split_text(text: str, n=100, character=" ") -> List[str]:
+    """Split the text every ``n``-th occurence of ``character``"""
+    text = text.split(character)
+    return [character.join(text[i : i + n]).strip() for i in range(0, len(text), n)]
+
+
+def split_documents(documents: dict) -> dict:
+    """Split documents into passages"""
+    titles, texts = [], []
+    for title, text in zip(documents["title"], documents["text"]):
+        for passage in split_text(text):
+            titles.append(title)
+            texts.append(passage)
+    return {"title": titles, "text": texts}
+
+
+def embed(documents: dict, ctx_encoder: DPRContextEncoder, ctx_tokenizer: DPRContextEncoderTokenizerFast) -> dict:
+    """Compute the DPR embeddings of document passages"""
+    input_ids = ctx_tokenizer(
+        documents["title"], documents["text"], truncation=True, padding="longest", return_tensors="pt"
+    )["input_ids"]
+    embeddings = ctx_encoder(input_ids.to(device=device), return_dict=True).pooler_output
+    return {"embeddings": embeddings.detach().cpu().numpy()}
+
+
+def main(
+    tmp_dir: str,
+    num_proc=2,
+    batch_size=16,
+    rag_model_name="facebook/rag-sequence-nq",
+    dpr_ctx_encoder_model_name="facebook/dpr-ctx_encoder-single-nq-base",
+):
+
+    ######################################
+    print("Step 1 - Create the dataset")
+    ######################################
+
+    # The dataset needed for RAG must have three columns:
+    # - title (string): title of the document
+    # - text (string): text of a passage of the document
+    # - embeddings (array of dimension d): DPR representation of the passage
+
+    # Let's say you have documents in tab-separated csv files with columns "title" and "text"
+    csv_path = str(Path(__file__).parent / "test_data" / "my_knowledge_dataset.csv")
```

Comment:
Can we put `csv_path` also in the main with the default "test_data/my_knowledge_dataset.csv"? => probs better than hardcoding it here